### ISSUE TITLE
Running dotnet publish in a netcoreapp2.0 project fails because of missing solution

### ISSUE THREAD

**Issue by janpieterz** [Particular/NServiceBus.Persistence.Sql#180]
`MSBUILD : error : SqlPersistenceTask: SolutionDirectory '*Undefined*' does not exist. [C:\Development\FastPACE\Api\Api.csproj]`

Have not been able to test this on a solution quickly because it contains non-dotnet core projects too.

**Comment by SimonCropp** [comment]
@janpieterz can u try adding this to the offending csproj

```
  <PropertyGroup>
    <SolutionDir Condition="$(SolutionDir) == '' Or $(SolutionDir) == '*Undefined*'">..\</SolutionDir>
  </PropertyGroup>
```

with the `..\` walking up enough levels to match the solution directory

**Comment by janpieterz** [comment]
That makes it work, as expected! 
The fix going into beta0004?

**Comment by SimonCropp** [comment]
not exactly. we will be making SolutionDir optional. and u will only need the workaround when in the scenario that causes no SolutionDir to exist and you also use SolutionDir in the promotion path. https://docs.particular.net/persistence/sql/#script-creation-promotion 

but yes this should mean you do not need the hack in the next beta.

**Comment by SimonCropp** [comment]
@janpieterz 3.0.0-alpha0005 is on myget and should fix this for you

**Comment by janpieterz** [comment]
No worries! Just updated and confirmed here 😄

**Comment by SimonCropp** [comment]
@janpieterz thanks heaps. btw what other beta bits are you experimenting with? any other feedback would be greatly appreciated

**Comment by janpieterz** [comment]
@SimonCropp  NServiceBus, NServiceBus.Autofac, NServiceBus.Persistence.Sql, NServiceBus.Persistence.Sql.MsBuild, NServiceBus.Serilog, NServiceBus.SqlServer in two products.

Still running on Windows/net47 at this moment, waiting for a couple of other small dependencies (GitVersionTask, dbup, Dapper, Topshelf) to either be upgraded or confirmed that it works to move to .net core 2.0. Also got an issue with xunit not recognizing .net core app 2.0 tests.

Basically, everything on your end looks great on .net 4.7 😄 

Love the Sql Persistence as it so blissfully removes the NHibernate dependency, runs smooth, migrated one product fully to (Azure hosted) MSSQL last week, instead of Azure ServiceBus. 
Anything I can complain about I'll shout 😉

**Comment by SzymonPobiega** [comment]
Still sometimes happens when running tests via R#. Rebuilding seems to solve the problem.

**Comment by janpieterz** [comment]
Logs:

```
To prevent NuGet from restoring packages during build, open the Visual Studio Options dialog, click on the Package Manager node and uncheck 'Allow NuGet to download missing packages during build.'
1>------ Build started: Project: Api, Configuration: Debug Any CPU ------
2>------ Build started: Project: Service, Configuration: Debug Any CPU ------
1>C:\Users\Jan-PieterZoutewelle\.nuget\packages\nservicebus.persistence.sql.msbuild\3.0.0-beta0004\build\NServiceBus.Persistence.Sql.MsBuild.targets(28,5): error MSB4044: The "ScriptBuilderTask" task was not given a value for the required parameter "SolutionDirectory".
1>Done building project "Api.csproj" -- FAILED.
```

**Comment by SimonCropp** [comment]
Btw there is a new wave of beta packages. If u have time we appreciate any feedback

**Comment by janpieterz** [comment]
Will update tomorrow and if anything changes report back!

**Comment by SimonCropp** [comment]
turns out is was a bug on our side. fixed now

**Comment by SimonCropp** [comment]
i accidentally fixed this on the postgre branch. so i moved this issue to 3.1. raised a duplicate #216 and assigned it to 3.0.

**Comment by janpieterz** [comment]
@SimonCropp looks like it's creeped back in somehow:

```
MSBUILD : error : SqlPersistenceTask: SolutionDirectory '*Undefined*' does not exist. [D:\a\1\s\Api\Api.csproj]
```

```
    <PackageReference Include="NServiceBus.Azure.Transports.WindowsAzureServiceBus" Version="8.0.0-rc0001" />
    <PackageReference Include="NServiceBus.Persistence.Sql" Version="4.0.0-rc0001" />
    <PackageReference Include="NServiceBus.Persistence.Sql.MsBuild" Version="4.0.0-rc0001" />
    <PackageReference Include="NServiceBus.Serilog" Version="4.0.0-beta001" />
```

**Comment by bording** [comment]
There is a regression here. I've raised #257 to fix it.
